cmd/oracled/cmd/get_oracle_key.go

```diff
@@ -28,18 +29,40 @@ func getOracleKeyCmd() *cobra.Command {
 
 			uniqueID := svc.EnclaveInfo().UniqueIDHex()
 			oracleAddress := svc.OracleAcc().GetAddress()
-			oracleRegistration, err := svc.QueryClient().GetOracleRegistration(ctx, uniqueID, oracleAddress)
+
+			from, err := cmd.Flags().GetString(flags.FlagFromOracleRegistrationOrUpgrade)
 			if err != nil {
-				return fmt.Errorf("failed to get oracle registration: %w", err)
+				return err
 			}
 
-			if len(oracleRegistration.EncryptedOraclePrivKey) == 0 {
-				return fmt.Errorf("the encrypted oracle private key has not set yet. please try again later")
+			if from == "registration" {
```

Comment:
It'd be safe to define these magic strings as const. Sorry that Go doesn't have enum.
```go
const (
  fromRegistration = "registration"
  fromUpgrade = "upgrade" 
)
```

Or, you can emulate the enum like other programming langs.
```go
type oracleKeyFetchMode string

const (
  fromRegistration oracleKeyFetchMode = "registration"
  fromUpgrade oracleKeyFetchMode = "upgrade" 
)

func parseOracleKeyFetchMode(str string) (oracleKeyFetchMode, error) {
  switch str {
  case: fromRegistration
     return fromRegistration
  ...
  default:
      return fmt.Errorf("invalid...")
  }
}
```
(I didn't check the syntax of these codes).

Reply:
Thanks. As you commented, I changed the magic string to const.